metricflow-semantics/metricflow_semantics/naming/linkable_spec_name.py

```diff
@@ -29,6 +27,18 @@ class StructuredLinkableSpecName:
     time_granularity_name: Optional[str] = None
     date_part: Optional[DatePart] = None
 
+    def __init__(  # noqa: D107
+        self,
+        entity_link_names: Tuple[str, ...],
+        element_name: str,
+        time_granularity_name: Optional[str] = None,
+        date_part: Optional[DatePart] = None,
+    ) -> None:
+        self.entity_link_names = tuple([entity_link_name.lower() for entity_link_name in entity_link_names])
+        self.element_name = element_name.lower()
+        self.time_granularity_name = time_granularity_name.lower() if time_granularity_name else None
```

Comment:
Careful, the time_granularity_name can be defined in an enum, if we alter it then calling TimeGranularity(time_granularity_name) won't work. More importantly, the checks we have elsewhere to see if a given granularity is custom or not won't work for those values either.

Let's move the assertion about is_standard_name on the TimeGranularity enumeration into this initializer (unless there's a way to do that assertion in the enum class itself). Then we can delete the other dataclass elements.